spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +65,27 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/{gatewayId}/apiKeys")
+    public ResponseEntity<List<ApiKey>> findApiKeyListByGatewayId(@PathVariable("gatewayId") String gatewayId) {
+        Proxy proxy = null;
+        List<ApiKey> apiKeyList;
+        String apiKeyListCreationFailedMessage = "Creation of API key list for requesting gateway failed. " +
+                "This concerns the gateway with the ID {}. ";
+        try {
+            proxy = proxyRepository.findById(gatewayId);
+            apiKeyList = proxy.getServiceApis().stream().flatMap(serviceApi -> serviceApi.getApiKeys()
+                    .stream()).collect(Collectors.toList());
+            return new ResponseEntity<>(apiKeyList, HttpStatus.OK);
+        } catch (NullPointerException e){
+            if (proxy == null)
+                log.info(apiKeyListCreationFailedMessage + "The gateway is not known.", gatewayId);
+            else if (proxy.getServiceApis() == null)
+                log.info(apiKeyListCreationFailedMessage + "The gateway does not provide any services.", gatewayId);
```

Comment:
I would assume that if the gateway does not provide any services, then `getServiceApis()` should return an empty collection, but not `null`. In any case, the reply to the gateway should be an "empty list" in my opinion, not an error message.

Reply:
I agree. It is now implemented as you suggested.